packages/react/src/react/components/primitives/tanstack-table.tsx

```diff
@@ -41,6 +42,28 @@ export interface TanstackTableProps<T> {
   }
 }
 
+const getCommonPinningClassesAndStyle = (column: Column<any>) => {
+  const isPinned = column.getIsPinned()
+  const isLastLeftPinnedColumn = isPinned === 'left' && column.getIsLastColumn('left')
+  const isFirstRightPinnedColumn = isPinned === 'right' && column.getIsFirstColumn('right')
+
+  const className = clsx(
+    isPinned ? 'sticky z-[1]' : 'relative',
+    isLastLeftPinnedColumn && 'shadow-[inset_-4px_0_4px_-4px_gray]',
+    isFirstRightPinnedColumn && 'shadow-[inset_4px_0_4px_-4px_gray]'
```

Comment:
Don't we have shadow token in design token?

Reply:
only want shadow to be one-sided